langchain/embeddings/openai.py

```diff
@@ -192,66 +192,63 @@ def _get_len_safe_embeddings(
         embeddings: List[List[float]] = [[] for _ in range(len(texts))]
         try:
             import tiktoken
+        except ImportError:
+            raise ImportError(
+                "Could not import tiktoken python package. "
+                "This is needed in order to for OpenAIEmbeddings. "
+                "Please install it with `pip install tiktoken`."
+            )
 
-            tokens = []
```

Comment:
bit hard to tell exactly what's going on here but guessing you just took all the downstream code out of the try/except without any functional changes?

Reply:
i think some classes perform the check at instantiation, so in those cases it's fine not to check within an instance method (since you would've had to successfully instantiate the object to get there)

Reply:
But then why this import is inside the code not at the file beginning? I'm still not comfortable with this idea. But the worst scenario we still get the same ImportError but without a clear explanation of how the missed package should be installed ("... use `pip install <package-name>`"). It is not a big deal.